src/main/java/com/avispl/symphony/dal/avdevices/camera/panasonic/awue150/CameraPanasonicAWUE150Communicator.java

```diff
@@ -0,0 +1,1779 @@
+				}
+			} catch (Exception e) {
+				logger.error(String.format("error while deserializing live camera information: %s", e.getMessage()), e);
+				updateFailedMonitor(DevicesMetricGroup.CAMERA_LIVE_INFO.getName(), "error while deserializing system information");
+			}
+			if (!fields.isEmpty()) {
+				// put live camera info to cache
+				cachedLiveCameraInfo.setGain(Gain.getByAPIValue(fields));
+				cachedLiveCameraInfo.setAwbMode(AWBMode.getByAPIValue(fields));
+				cachedLiveCameraInfo.setShutter(Shutter.getByAPIValue(fields));
+				cachedLiveCameraInfo.setNighDayFilter(NightDayFilter.getByAPIValue(fields));
+				cachedLiveCameraInfo.setAutoFocus(AutoFocus.getByAPIValue(fields));
+				cachedLiveCameraInfo.setFocusADJWithPTZ(FocusADJWithPTZ.getByAPIValue(fields).getUiName());
+				cachedLiveCameraInfo.setOperationLock(OperationLock.getByAPIValue(fields));
+				cachedLiveCameraInfo.setPowerStatus(PowerStatus.getByAPIValue(fields));
+				cachedLiveCameraInfo.setPowerOnPosition(PowerOnPosition.getByAPIValue(fields));
+				cachedLiveCameraInfo.setOutputFormat(OutputFormat.getByAPIValue(fields));
+				cachedLiveCameraInfo.setIrisMode(IrisMode.getByAPIValue(fields));
+				cachedLiveCameraInfo.setPresetEntitiesStatus(
+						mapPresetEntityToListPreset(cachedLiveCameraInfo.getPreset01To40(), cachedLiveCameraInfo.getPreset41To80(), cachedLiveCameraInfo.getPreset81To100()));
+
+				// populate live camera info
+				stats.put(DeviceInfoMetric.OUTPUT_FORMAT.getName(), cachedLiveCameraInfo.getOutputFormat().getUiName());
+				stats.put(DeviceInfoMetric.OPERATION_LOCK.getName(), cachedLiveCameraInfo.getOperationLock().getUiName());
+				stats.put(DeviceInfoMetric.CAMERA_TITLE.getName(), getDefaultValueForNullData(fields.get(Command.TITLE_RESPONSE), DeviceConstant.NONE));
+				stats.put(DeviceInfoMetric.FAN1_STATUS.getName(), Fan1Status.getByAPIValue(fields).getUiName());
+				stats.put(DeviceInfoMetric.FAN2_STATUS.getName(), Fan2Status.getByAPIValue(fields).getUiName());
+				stats.put(DeviceInfoMetric.ERROR_INFORMATION.getName(), ErrorInformation.getByAPIValue(fields).getUiName());
+				stats.put(DeviceInfoMetric.ERROR_STATUS_INFO.getName(), ErrorStatusInformation.getByAPIValue(fields).getUiName());
+				stats.put(DeviceInfoMetric.POWER_STATUS.getName(), cachedLiveCameraInfo.getPowerStatus().getUiName());
+				stats.put(DeviceInfoMetric.UHD_CROP.getName(), UHDCrop.getByAPIValue(fields).getUiName());
+			}
+		} else {
+			updateFailedMonitor(DevicesMetricGroup.CAMERA_LIVE_INFO.getName(), "error while getting camera live information");
+		}
+	}
+
+	/**
+	 * This method is used to retrieve model name information by send get request to "http://10.8.53.221/cgi-bin/model_serial"
+	 *
+	 * @param stats store all statistics
+	 * When the response is null or empty, the failedMonitor is going to update and exception is not populated
+	 */
+	private void retrieveModelName(Map<String, String> stats) throws FailedLoginException {
```

Comment:
javadoc does not include information about the Exception in method signature

Reply:
I fixed it